scripts/process_diagnostic_messages.rs

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env rust-script
+//!
+//! ```cargo
+//! [dependencies]
+//! serde = { version = "1.0", features = ["derive"] }
+//! serde_json = "1.0"
+//! pathdiff = "0.2.1"
+//! regex = "1.5.4"
+//! ```
+
+use regex::Regex;
+use serde::Deserialize;
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+enum DiagnosticCategory {
+    Error,
+    Message,
+    Suggestion,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+struct DiagnosticMessageSpec {
+    category: DiagnosticCategory,
+    code: u32,
+}
+
+fn main() {
+    let args = env::args().collect::<Vec<_>>();
+    if args.len() < 2 {
+        println!("Usage:");
+        println!("\t./scripts/process_diagnostic_messages.rs <diagnostic-json-input-file>");
+        return;
+    }
+    let input_file_path = Path::new(&args[1])/*.replace(/\\/g, "/")*/;
+    println!("Reading diagnostics from {}", input_file_path.display());
+    let input_str = fs::read_to_string(input_file_path).expect("Couldn't read input file");
+    let diagnostic_messages_json: HashMap<String, DiagnosticMessageSpec> =
+        serde_json::from_str(&input_str).expect("Couldn't parse input file as JSON");
+    let diagnostic_messages = diagnostic_messages_json;
+    let output_files_dir = input_file_path
+        .parent()
+        .expect("Couldn't get input file path parent");
+    let this_file_path_rel = pathdiff::diff_paths(
```

Comment:
Per [this answer](https://stackoverflow.com/a/46382269/732366), [`pathdiff`](https://docs.rs/pathdiff/latest/pathdiff/index.html) lets you get a relative path between two absolute paths (which mimics how the Typescript version was using NodeJS's `path.relative()`)